form/form/form_source_type_registry.cpp

```
@@ -0,0 +1,87 @@
+      throw std::runtime_error("Cannot register empty FORM product type name");
+    }
+    if (!reader_fn) {
+      throw std::runtime_error("Cannot register FORM product type with empty reader function");
+    }
+
+    std::lock_guard<std::mutex> lock(form_type_registry_mutex());
+    mutable_form_type_registry()[std::move(product_type)] =
+      form_source_type_entry{std::move(type), &cpp_type, std::move(reader_fn)};
+  }
+
+  form_source_type_entry const* find_form_product_type(std::string const& product_type)
```

Cppcheck (2.21.0)

[style] The function 'find_form_product_type' is never used.

(unusedFunction)